quinn/src/udp.rs

```diff
@@ -53,11 +54,11 @@ impl UdpSocket {
         meta: &mut [RecvMeta],
     ) -> Poll<io::Result<usize>> {
         debug_assert!(!bufs.is_empty());
-        ready!(self.io.poll_read_ready(cx, mio::Ready::readable()))?;
+        let mut guard = ready!(self.io.poll_read_ready(cx))?;
         match self.io.get_ref().recv_ext(bufs, meta) {
             Ok(n) => Poll::Ready(Ok(n)),
             Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
-                self.io.clear_read_ready(cx, mio::Ready::readable())?;
+                guard.clear_ready();
                 Poll::Pending
             }
             Err(e) => Poll::Ready(Err(e)),
```

Comment:
I recommend that you rewrite this to use the [`try_io`][1] method from the guard together with a loop like in the example on `AsyncFd`, because the `clear_ready` method no longer takes a context, and thus it no longer registers the current waker for wakeup.

[1]: https://docs.rs/tokio/1/tokio/io/unix/struct.AsyncFdReadyGuard.html#method.try_io

Reply:
Done. This seems quite a bit nicer! `clear_ready()` not taking a context seems like a footgun, maybe needs more docs?